GoogleCloudPlatform/gcsfuse
# add prometheus metrics support

### Description

Add Prometheus metrics support. Add a new flag `prometheus-port` for users to specify prometheus metrics endpoint port.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Done

- Build the code using the command `go run tools/build_gcsfuse/main.go . . v2.0.0`.
- Mount a bucket: `./bin/gcsfuse --prometheus-port 8080 <bucket-name> <mount-path>`.
- Pull the metrics: `curl http://localhost:8080/metrics`
- See the output example in the docs/metrics.md.

3. Unit tests - NA
4. Integration tests - NA
